pkg/portfwd/client.go

```diff
@@ -24,27 +25,15 @@ func HandleTCPConnection(ctx context.Context, client *guestagentclient.GuestAgen
 		return
 	}
 
-	g, _ := errgroup.WithContext(ctx)
-
 	rw := &GrpcClientRW{stream: stream, id: id, addr: guestAddr}
-	g.Go(func() error {
-		_, err := io.Copy(rw, conn)
-		if errors.Is(err, io.EOF) {
-			return nil
-		}
```

Comment:
Looking at io.Copy implementation:

```
func copyBuffer(dst Writer, src Reader, buf []byte) (written int64, err error) {
	// If the reader has a WriteTo method, use it to do the copy.
	// Avoids an allocation and a copy.
	if wt, ok := src.(WriterTo); ok {
		return wt.WriteTo(dst)
	}
	// Similarly, if the writer has a ReadFrom method, use it to do the copy.
	if rf, ok := dst.(ReaderFrom); ok {
		return rf.ReadFrom(src)
	}
	if buf == nil {
		size := 32 * 1024
		if l, ok := src.(*LimitedReader); ok && int64(size) > l.N {
			if l.N < 1 {
				size = 1
			} else {
				size = int(l.N)
			}
		}
		buf = make([]byte, size)
	}
	for {
		nr, er := src.Read(buf)
		if nr > 0 {
			nw, ew := dst.Write(buf[0:nr])
			if nw < 0 || nr < nw {
				nw = 0
				if ew == nil {
					ew = errInvalidWrite
				}
			}
			written += int64(nw)
			if ew != nil {
				err = ew
				break
			}
			if nr != nw {
				err = ErrShortWrite
				break
			}
		}
		if er != nil {
			if er != EOF {
				err = er
			}
			break
		}
	}
	return written, err
}
```

It will not return io.EOF from the copy loop - but it may delegate the copy to ReadFrom() or WriteTo(), which can be implemented by the actual types we copy. I don't think we have any guarantee that we will never get io.EOF.

So it may be better to keep this if we don't want to report it as an error.

Reply:
`ReadFrom` is not permitted to return EOF:
https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/io/io.go;l=186

`WriteTo` returns "Any error encountered during the write":
https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/io/io.go;l=197
EOF does not occur on write, only on read.

Reply:
Good about ReadFrom - so we can remove the io.EOF checks than.